File: public/dsa/searching/square_root/naive/square_root.py
class Solution:
    def searchRoot(self, num: int) -> int:
        if num == 0 or num == 1:
            return num
        for i in range(1, num + 1):
            sq = i * i
            if sq == num:
                return i
            elif sq > num:
                return i - 1
        return -1


# tests-start
sol = Solution()
print(sol.searchRoot(16))
print(sol.searchRoot(12))
# tests-end

'''output
4
3
'''

File: public/dsa/searching/square_root/naive/square_root.cpp
#include <bits/stdc++.h>
using namespace std;

int searchRoot(int num) {
    if (num == 0 || num == 1) {
        return num;
    }
    for (int i = 1; i <= num; i++) {
        int sq = i * i;
        if (sq == num) {
            return i;
        } else if (sq > num) {
            return i - 1;
        }
    }
    return -1;
}

// tests-start
int main() {
    cout << searchRoot(16) << endl;
    cout << searchRoot(12) << endl;
}
// tests-end

/*output
4
3
*/